python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1321,7 +1321,7 @@ def get_subset_selection_for_context(
 
     # note that this will always be true if checks are disabled
     if selected_checks_target_selected_assets and all_checks_included_for_selected_assets:
-        selected_dbt_resources = selected_dbt_non_test_resources + selected_dbt_tests
```

Comment:
glad I caught this- bug from https://github.com/dagster-io/dagster/pull/ that had us still listing checks explicitly. Tested by https://github.com/dagster-io/dagster/blob/johann/03-28-Default_dbt_asset_checks_on/python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_resources_v2.py#L375 now that checks are on by default

Reply:
The pull request you linked in broken -- could you link the correct one?

Reply:
oops, https://github.com/dagster-io/dagster/pull/20568

Reply:
Could you add an explicit test in `test_asset_checks.py`? One that would have failed master without this change.

We're relying on the implicit assumption that asset checks is enabled (which is why we caught this in `test_resources_v2.py` in this PR. 

We should make our implicit assumption explicit in the test case. [1]

Reply:
moved the bugfix to a separate pr with explicit tests: https://github.com/dagster-io/dagster/pull/20944